lib/reversetunnel/conn_metric.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package reversetunnel
+
+import (
+	"net"
+	"sync"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type dialType string
+
+const (
+	// direct is a direct dialed connection.
+	direct dialType = "direct"
+	// peer is a connection established through a peer proxy.
+	peer dialType = "peer"
+	// tunnel is a connection established over a local reverse tunnel initiated
+	// by a client.
+	tunnel dialType = "tunnel"
+	// peerTunnel is a connection established over a local reverse tunnel
+	// initiated by a peer proxy.
+	peerTunnel dialType = "peer-tunnel"
+)
+
+// metricConn reports metrics for reversetunnel connections.
+type metricConn struct {
+	net.Conn
+	clock clockwork.Clock
+
+	// start is the time since the last state was reported.
+	start    time.Time
+	firstUse sync.Once
```

Comment:
Nit: we could maybe rename all instances of "first use" to "first read", or something else that better represents a connection `state` (the histogram label) like "established".

Reply:
+1 for `first read`
You can also change to another variation and add the `Once` suffix which seems common on the codebase